modules/behavior/draw_way.js

```diff
@@ -412,6 +413,81 @@ export function behaviorDrawWay(context, wayID, mode, startGraph) {
         });
     };
 
+    /** see PR #8671 */
+    function followMode() {
+        if (_didResolveTempEdit) return;
+
+        try {
+
+            // get the last 2 added nodes.
+            // check if they are both part of only oneway (the same one)
+            // check if the ways that they're part of are the same way
+            // find index of the last two nodes, to determine the direction to travel around the existing way
+            // add the next node to the way we are drawing
+
+            // if we're drawing an area, the first node = last node.
+            const isDrawingArea = _origWay.nodes[0] === _origWay.nodes.slice(-1)[0];
+
+            const [secondLastNodeId, lastNodeId] = _origWay.nodes.slice(isDrawingArea ? -3 : -2);
+
+            if (!lastNodeId || !secondLastNodeId) {
+                context.ui().flash
+                    .duration(4000)
+                    .iconName('#iD-icon-no')
+                    .label(t('operations.follow.error.needs_more_initial_nodes'))();
+                return;
+            }
+
+            const lastNodesParents = startGraph.parentWays(startGraph.entity(lastNodeId));
+            const secondLastNodesParents = startGraph.parentWays(startGraph.entity(secondLastNodeId));
+
+            if (lastNodesParents.length !== 1 || secondLastNodesParents.length !== 1) {
```

Comment:
The current behavior is: code is able to deduce which path to follow if last two nodes have a single parent.

I think we could also deduce which path to follow in situation where only one of the last two nodes has a single parent,

Reply:
:+1: fixed